Fix resolving of variables for che commands

RomanNikitenko opened the pull request:
### What does this PR do?
Allows to resolve variables for che commands before execution.
For example you can define command in your devfile:
```
- name: copy content
  actions:
  - type: exec
    component: che-dev
    command: >
              cp ${workspaceFolder}/.theia/tasks.json ${workspaceFolder}/cheCommands.json
    workdir: /projects
```
and `${workspaceFolder}` will be replaced by `/projects` for example, or use `${file}` to get path to 
 the file opened in the current editor

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14031

### How to test
Use the component in your devfile

```
- 
    alias: theia-editor
    reference: >-
      https://raw.githubusercontent.com/RomanNikitenko/che-plugin-registry/master/v3/plugins/eclipse/che-theia/custom/meta.yaml
    type: cheEditor

```
The image contains:
- che-7.0.0 theia branch
- current 7.0.0 branch of che-theia + changes from the PR
 
Use a command with variables in your devfile, the simple example you can see above.

AndrienkoAleksandr commented:
I want to test this branch.

RomanNikitenko commented:
> I want to test this branch.

@AndrienkoAleksandr 
I'd appreciate it!
Do you need some help from my side for it?

azatsarynnyy commented:
It would be nice to add resolving the variables in `target.workingDir` property as well. It's very helpful.

RomanNikitenko commented:
> It would be nice to add resolving the variables in `target.workingDir` property as well. It's very helpful.

maybe it's implemented on machine-exec level, because I see that variables [are used](https://github.com/eclipse/che-devfile-registry/blob/b4a78f9a75e67b4beae8a67865ed612812d891d1/devfiles/python/devfile.yaml#L30) for `workingDir` in devfiles for samples.

AndrienkoAleksandr commented:
> maybe it's implemented on machine-exec level, because I see that variables are used for workingDir in devfiles for samples.

che-machine-exec doesn't do such things.

azatsarynnyy commented:
`${CHE_PROJECTS_ROOT}` it's not Theia's substitution variable but the environment variable

RomanNikitenko commented:
@AndrienkoAleksandr @azatsarynnyy 
thank you for clarification!

RomanNikitenko commented:
@AndrienkoAleksandr @azatsarynnyy @olexii4 
I added resolving of the variables for `workingDir` property after your review.

btw: I updated the images for testing (No pressure, just info for case if somebody wants to try it).

AndrienkoAleksandr commented:
Tested, works fine.

RomanNikitenko commented:
@slemeur 
Can I merge the PR into 7.0.0 branch?
thank you!

slemeur commented:
I'm +1 for it.
@l0rd
